packages/wallet/backend/src/auth/service.ts

```diff
@@ -4,27 +4,55 @@ import addSeconds from 'date-fns/addSeconds'
 import type { Env } from '@/config/env'
 import type { Session } from '@/session/model'
 import type { UserService } from '@/user/service'
+import { getRandomToken, hashToken } from '@/utils/helpers'
+import { EmailService } from '@/email/service'
+import { Logger } from 'winston'
 
 interface AuthorizeArgs {
   email: string
   password: string
 }
+
+interface SignupArgs extends AuthorizeArgs {}
+
 interface AuthorizeResult {
   user: User
   session: Session
 }
 
 interface IAuthService {
   authorize(args: AuthorizeArgs): Promise<AuthorizeResult>
+  signup(args: SignupArgs): Promise<User>
 }
 interface AuthServiceDependencies {
   userService: UserService
+  emailService: EmailService
+  logger: Logger
   env: Env
 }
 
 export class AuthService implements IAuthService {
   constructor(private deps: AuthServiceDependencies) {}
 
+  async signup({ email, password }: SignupArgs): Promise<User> {
+    const token = getRandomToken()
+
+    const user = await this.deps.userService.create({
+      email,
+      password,
+      verifyEmailToken: hashToken(token)
+    })
+
+    await this.deps.emailService.sendVerifyEmail(email, token).catch((e) => {
+      this.deps.logger.error(
+        `Error on sending verify email for user ${user.email}`,
+        e
+      )
+    })
```

Comment:
I don't think that's the correct flow. We should:
1. do the email verification - we do not create the user if the email checks do not pass
2. create the user
3. send the email